closed/src/java.base/share/classes/sun/security/ec/NativeECDHKeyAgreement.java

```diff
@@ -150,18 +173,20 @@ protected Key engineDoPhase(Key key, boolean lastPhase)
             throw new IllegalStateException
                 ("Only two party agreement supported, lastPhase must be true");
         }
-        if (!(key instanceof PublicKey)) {
+        if (!(key instanceof ECPublicKey ecKey)) {
             throw new InvalidKeyException
                 ("Key must be an instance of PublicKey");
         }
-        /* attempt to translate the key if it is not an ECKey */
-        ECKey ecKey = ECKeyFactory.toECKey(key);
+
         if (ecKey instanceof ECPublicKeyImpl keyImpl) {
             this.publicKey = keyImpl;
 
             int keyLenBits = this.publicKey.getParams().getCurve().getField().getFieldSize();
             this.secretLen = (keyLenBits + 7) >> 3;
 
+            // Validate public key.
+            validate(privateKeyOps, this.publicKey);
```

Comment:
In the head stream, this validation occurs before line 181. I think we want the same sequence here.

Reply:
Moved.